recipes/systemc/2.3.3/conanfile.py

```diff
@@ -0,0 +1,113 @@
+        "enable_pthreads": False
+    }
+    generators = "cmake"
+    exports_sources = "patches/**"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos build not supported")
+
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("The compilation of SystemC as a "
+                                            "DLL on Windows is currently not "
+                                            "supported")
+
+        if tools.valid_min_cppstd(self, "17"):
+            raise ConanInvalidConfiguration(
+                "C++ Standard %s not supported by SystemC" %
+                self.settings.compiler.cppstd)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("systemc-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["DISABLE_ASYNC_UPDATES"] = \
+            self.options.disable_async_updates
+        self._cmake.definitions["DISABLE_COPYRIGHT_MESSAGE"] = \
+            self.options.disable_copyright_msg
+        self._cmake.definitions["DISABLE_VIRTUAL_BIND"] = \
+            self.options.disable_virtual_bind
+        self._cmake.definitions["ENABLE_ASSERTIONS"] = \
+            self.options.enable_assertions
+        self._cmake.definitions["ENABLE_IMMEDIATE_SELF_NOTIFICATIONS"] = \
+            self.options.enable_immediate_self_notifications
+        self._cmake.definitions["ENABLE_PTHREADS"] = \
+            self.options.enable_pthreads
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy("NOTICE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = ["systemc"]
+        # FIXME: cmake generates SystemC::systemc target, not SystemC::SystemC
+        self.cpp_info.names["cmake_find_package"] = "SystemC"
+        self.cpp_info.names["cmake_find_package_multi"] = "SystemC"
+
+        if self.settings.os == "Linux":
```

Comment:
```suggestion
        if self.settings.os == "Linux" and self.options.enable_pthreads:
```

https://github.com/accellera-official/systemc/blob/604182509559ae42c34e878f508bae7c18abbbd6/CMakeLists.txt#L534

Reply:
Note that pthreads is still used in case of c++ 98 on linux even if `enable pthreads` is false.
Setting `enable pthreads` to false only indicates that SystemC processes should not use pthreads, the remainder of the library can still use pthreads.

https://github.com/accellera-official/systemc/blob/b7e5ef795bb46a73b950355cc7e4484e63c6c068/src/sysc/communication/sc_host_mutex.h#L92